script.artistslideshow/resources/lib/artistslideshow.py

```diff
@@ -0,0 +1,1240 @@
+            for image in self.IMAGES:
+                if self.IMAGEADDED or self.IMAGESCLEARED:
+                    lw.log( ['image list changed, resetting loop'] )
+                    break
+                if not image == last_image or len( self.IMAGES ) == 1:
+                    if not image == last_image:
+                        self._set_property( 'ArtistSlideshow.Image', image )
+                        last_image = image
+                    self._wait( wait_time=self.DELAY, sleep_time=self.SLIDESHOWSLEEP )
+                if self._check_for_quit():
+                    break
+        lw.log( ['slideshow thread stopping'] )
+
+
+    def _check_for_quit( self ):
+        cmd = ''
+        with self.QUEUELOCK:
+            if not self.WORKQUEUE.empty():
+                cmd = self.WORKQUEUE.get()
+            if cmd == 'quit':
+                self.SHOW = False
+        return not self.SHOW
+
+
+    def _set_property( self, property_name, value='' ):
+        try:
+          self.WINDOW.setProperty( property_name, value )
+          lw.log( ['%s set to %s' % (property_name, value)] )
+        except Exception as e:
+          lw.log( ['Exception: Could not set property %s to value %s' % (property_name, value), e] )
+
+
+    def _wait( self, wait_time=1, sleep_time=1 ):
+        waited = 0
+        while waited < wait_time:
+            xbmc.sleep( sleep_time*1000 )
+            waited = waited + sleep_time
+            if self._check_for_quit():
+                return
+
+
+
+class Main( object ):
+
+    def __init__( self ):
+        self._parse_argv()
+        self._init_window()
+        if self._get_infolabel( self.ARTISTSLIDESHOWRUNNING ) == 'True' and not self.RUNFROMSETTINGS:
+            lw.log( ['script already running'] )
+        else:
+            self._upgrade_settings()
+            self._get_settings()
+            self._init_vars()
+            self._make_dirs()
+            self._upgrade()
+            if self._run_from_settings():
+                return
+            self._set_property( 'ArtistSlideshowRunning', 'True' )
+            if not xbmc.Player().isPlayingAudio() and self._get_infolabel( self.EXTERNALCALL ) == '':
+                lw.log( ['no music playing'] )
+                if not self.DAEMON:
+                    self._set_property( 'ArtistSlideshowRunning' )
+                    self._set_property( 'ArtistSlideshow.Image' )
+            else:
+                lw.log( ['first song started'] )
+                if not xbmc.Monitor().waitForAbort( 1 ): # it may take some time for Kodi to read the tag info after playback started
+                    self._slideshow_thread_start()
+                    self._use_correct_artwork()
+                    self._trim_cache()
+                else:
+                    self._set_property( 'ArtistSlideshowRunning' )
+            sleeping = False
+            while not xbmc.Monitor().abortRequested() and self._get_infolabel( self.ARTISTSLIDESHOWRUNNING ) == 'True':
```

Comment:
probably a good ideia do create monitor and player objects and define them as member variables on init. Otherwise you're creating them all the time (which is expensive).